### DISCUSSION TITLE
The documented Valve script issue (GPIO out always HIGH)

### DISCUSSION THREAD

**Discussion by djaytek** [RavenSystem/esp-homekit-devices#2025] [MEPLHAA Scripts]
Hi, my first time using HAA and I tried creating a valve switch with the default documented script on ESP8266 (nodemcu).
https://github.com/RavenSystem/esp-homekit-devices/wiki/Water-Valve

I succesfully installed HAA and add the script and add the accessory to Homekit, these were very simple and easy.

But when I power the ESP8266, it triggers the relay to ON and I can't change the state after that.

What am I missing? I tried adding Pull-up/down settings, actions on boot, initial state but none of them helped. The GPIO out is always HIGH and do not change by any way (from the Home APP). The app properly shows to turn on or off the valve but in reality the Relay is always pulled and in HIGH state after powering up the nodemcu.

Relay module is connected to GPIO12 (D6 on the nodemcu board)

Here is my MEPLHAA script at the moment with latest modifications:

{
  "c": {
    "io": [ [[12,13],2,0,0], [[0],6] ],
    "l": 13, "b": [{ "g": 0, "t": 5 }]
  },
  "a": [{
    "t": 20,
    "0": { "r": [{ "g": 12, "v": 0 }] },
    "1": { "r": [{ "g": 12, "v": 1 }] },
    "b": [{ "g": 0, "t": 1 }],
    "s": 0,
    "w": 0,
    "d": 60,
    "xa": 0
  }]
}

**Comment by djaytek** [comment]
OK, I finally had some control after modifying the output mode, from:
2 | Output only
to:
3 | Output only with open-drain

But the function is inverted of what I need.

On home app when the device is off, the relay is on state (pulled up)
When device is on, relay is off state.

{
"c": {
"io": [ [[12,13],3,0,0], [[0],6] ],
"l": 13, "b": [{ "g": 0, "t": 5 }]
},
"a": [{
"t": 20,
"0": { "r": [{ "g": 12, "v": 0 }] },
"1": { "r": [{ "g": 12, "v": 1 }] },
"b": [{ "g": 0, "t": 1 }],
"s": 0,
"w": 0,
"d": 60
}]
}

**Comment by djaytek** [comment]
After a lot of tries, I somehow managed to make some progress but it is not optimal solution.
I internet the functions in the script. But now upon booting, the relay is trigged for a moment.

{
"c": {
"io": [ [[12,13],3], [[0],6] ],
"l": 13, "b": [{ "g": 0, "t": 5 }]
},
"a": [{
"t": 20,
"0": { "r": [{ "g": 12, "v": **1** }] },
"1": { "r": [{ "g": 12, "v": **0**}] },
"b": [{ "g": 0, "t": 1 }],
"s": 0,
"w": 0,
"d": 60
}]
}

**Reply by xrust83** [reply]
You need to understand the structure of MELPHAA, letters are assigned to certain gpios in or commands. The script in the example is designed for sonoff basic or TH, since they have 13 LEDs connected to the GPIO. Because you are trying to cross a bulldog and a rhinoceros, no one is going to help you. If you have DIYs, at least provide a scheme of work
